app/src/main/java/org/wikipedia/createaccount/CreateAccountActivity.kt

```diff
@@ -42,10 +42,10 @@ class CreateAccountActivity : BaseActivity() {
     private lateinit var binding: ActivityCreateAccountBinding
     private lateinit var captchaHandler: CaptchaHandler
     private lateinit var createAccountEvent: CreateAccountEvent
-    private val disposables = CompositeDisposable()
     private var wiki = WikipediaApp.instance.wikiSite
     private var userNameTextWatcher: TextWatcher? = null
-    private val userNameVerifyRunnable = UserNameVerifyRunnable()
+    private var verifyUserNameJob: Job? = null
```

Comment:
Can we move the `Job` to the ViewModel class and cancel the job before verifying the username?

Reply:
Yes, I've moved the job to the ViewModel, and now the `doOnTextChanged` method only calls `viewModel.verifyUserName(text)`